packages/core/src/withRetries.ts

```diff
@@ -1,28 +1,28 @@
-import type { API, SearchQuery, SearchResult } from "@nosto/nosto-js/client"
+import type { SearchQuery, SearchResult } from "@nosto/nosto-js/client"
 import { logger } from "@utils/logger"
 
-import { SearchOptions } from "./types"
+import { HitDecorator, SearchFn, SearchOptions } from "./types"
 import { delay } from "./utils/delay"
 
-export async function searchWithRetries(
-  api: API,
+export async function searchWithRetries<HD extends readonly HitDecorator[]>(
   query: SearchQuery,
-  options: SearchOptions<[]>
+  { maxRetries = 0, retryInterval = 0, ...options }: SearchOptions<HD>,
+  next: SearchFn<HD>
 ): Promise<SearchResult> {
-  const { maxRetries = 0, retryInterval = 1000 } = options
+  if (maxRetries <= 0) {
```

Comment:
Doesn't the `while(true)` block handle this case? Do we need this additional condition? This block is missing error handling too

Reply:
I can drop it